django/template/defaultfilters.py

```diff
@@ -28,10 +29,39 @@
 register = Library()
 
 
+def autoescape_aware(func):
+    """
+    Decorator for filters which should escape automatically escape text based
+    on whether or not autoescape is True.
+    """
+    def _dec(*args, **kwargs):
+        args = list(args)
+        try:
+            autoescape = kwargs['autoescape']
+        # if value is not defined, get default value
+        except KeyError:
+            for k, v in signature(func).parameters.items():
```

Comment:
I don't believe this block is covered by the test, is it?

Reply:
If you are referring to getting the default value, the first function in the test case uses the default value.

Reply:
Ah yes, you are correct sorry.